packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -0,0 +1,98 @@
+import * as cdk from '@aws-cdk/core';
+import { InputValidator } from './private/validation';
+import { CfnApplication } from './servicecatalogappregistry.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * A Service Catalog AppRegistry Application.
+ */
+export interface IApplication extends cdk.IResource {
+  /**
+   * The ARN of the application.
+   * @attribute
+   */
+  readonly applicationArn: string;
+
+  /**
+   * The ID of the application.
+   * @attribute
+   */
+  readonly applicationId: string;
+}
+
+/**
+ * Properties for a Service Catalog AppRegistry Application
+ */
+export interface ApplicationProps {
+  /**
+   * Enforces a particular physical application name.
+   */
+  readonly applicationName: string;
```

Comment:
BTW - you called this `displayName` in `Portfolio`. Any reason you went with a different name in `Application`?

Reply:
The actual cfn term in `Portfolio` is `DisplayName`, see [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-servicecatalog-portfolio.html).  For application it is just `name` hence `applicationName`.

Reply:
You don't want to make this consistent across the ServiceCatalog constructs that exhibit this behavior (display name + auto-generated ID)?